lib/battleship.rb

```diff
@@ -1,5 +1,38 @@
 require_relative "board"
 require_relative "player"
 
+#1) how to mark hit, instance variable?, just update board with "m"
 class BattleshipGame
+  attr_reader :board, :player
+
+  def initialize(player, board)
+    @player, @board = player, board
+  end
+
+  def play_turn
+    pos = player.get_play
+    attack(pos)
+  end
+
+  def play
+    play_turn until game_over?
+  end
+
+  def attack(pos)
+    board[pos] == :s ? board[pos] == :x : board[pos] ==  " M " #mark board with miss,
```

Comment:
`board[pos] == :x` almost !
`==` is an equality operator, here you need the assignation operator: `=`

You already defined the method `Board#[]` that allows you to read a value on the board given a position.

You will need to define the method `Board#[]=` that will assign a value to the board at the given position:

```ruby
# board.rb
def []=(pos, value)
  # ..
end
```

Reply:
that makes sense!  I added the following 
def []=(pos, value)
    row, col = pos
    grid[row][col] = value
  end